### ISSUE TITLE
Honeybadger.lambdaCallback does not pass event and context with nodejs4.3 runtime

### ISSUE THREAD

**Issue by WilHall** [honeybadger-io/honeybadger-node#10]
The following AWS Lambda function using the `nodejs4.3` runtime and any version of `honeybadger-node` does not pass the `event` and `context` objects to the wrapped handler function. The following function fails (written based on the `README` of this repo):

```
var Honeybadger = require('honeybadger');
Honeybadger.configure({
  apiKey: process.env.HONEYBADGER_API_KEY
});

function handler(event, context, callback) {
    console.log(event.foo);
    console.log(context.bar);
}

exports.handler = Honeybadger.lambdaHandler(handler);
```

This will cause an error like `Cannot read property '. . .' of undefined`; it appears that no arguments are passed to the function that is wrapped with `Honeybadger.lambdaHandler`, so `event` and `context` are not available.

**Comment by joshuap** [comment]
@WilHall Yeah, it looks like I overlooked something, sorry about that. I'll take a look at it today.

**Comment by joshuap** [comment]
After reading more about the changes in the nodejs4.3 runtime, it seems to me that the current `lambdaHandler` isn't even necessary. The event loop is emptied before each execution completes, which should solve the issue of waiting for errors to be reported before exiting (previously we had to call `context.fail` inside our notify callback).

So I'm thinking that you could do `exports.handler = Honeybadger.wrap(handler);` and get back a handler which reports errors and passes the original arguments through correctly. We'll need to fix #9 first, though.

**Comment by WilHall** [comment]
I'm currently doing `exports.handler = Honeybadger.wrap(handler);` with success (and on `v1.0.0` as a workaround for #9).

If you want to deprecate `Honeybadger.lambdaHandler` in favor of just calling `Honeybadger.wrap` on the `handler` function, I think that makes sense.

The only thing this would miss would be calls to the lambda handler `callback` with an error - Honeybadger would not catch those in this case, and the user would need to manually call `Honeybadger.notify` before calling the `callback` with an error. (E.g: http://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-handler.html#nodejs-prog-model-handler-callback)

One thing `Honeybadger.lambdaCallback` could do to automatically handle these would be to wrap the  `callback` handler parameter.
It could return a [wrapped function like Honeybadger.wrap does](https://github.com/honeybadger-io/honeybadger-node/blob/d40a0daf81508032b047c49b3469e2535f1bed2e/lib/honeybadger.js#L115) but call the original function with `arguments.slice(0, 2)` (`event`, `context`) verbatim, and with `arguments[2]` (`callback`) as a wrapped function, so that Honeybadger could automatically handle errors passed to the `callback`, too.

In the case that the `callback` parameter didn't exist, it would indicate that you were running with a pre-`nodejs4.3` runtime.
